src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -743,6 +747,45 @@ def _get_workspace_resource_id(
         # this parameter does not need validation
         return workspace_resource_id
 
+    def get_ip_families(self) -> Union[List[str], None]:
+        """IPFamilies used for the cluster network.
+
+        :return: List[str] or None
+        """
+        return self._get_list_attr('ip_families')
+
+    def get_pod_cidrs(self) -> Union[List[str], None]:
+        """Obtain the CIDR ranges used for pod subnets.
+
+        :return: List[str] or None
+        """
+        return self._get_list_attr('pod_cidrs')
+
+    def get_service_cidrs(self) -> Union[List[str], None]:
+        """Obtain the CIDR ranges for the service subnet.
+
+        :return: List[str] or None
+        """
+        return self._get_list_attr('service_cidrs')
+
+    def _get_list_attr(self, param_key) -> Union[List[str], None]:
+        param = self.raw_param.get(param_key)
+
+        if param and isinstance(param, str):
```

Comment:
In this way, we would convert empty string to `None`. If this is expected, please add a unit test case to cover this.

Reply:
what do we usually do in cases where the parameter is expected to be a list? convert to an empty list? `[]`?

Reply:
Looks like for other parameters we differentiate between omitted and empty string for list properties. Will update.